src/core_plugins/input_control_vis/public/control/control.js

```diff
@@ -100,26 +100,30 @@ export class Control {
 
   set(newValue) {
     this.value = newValue;
-    this._hasChanged = true;
     if (this.hasValue()) {
       this._kbnFilter = this.filterManager.createFilter(this.value);
     } else {
       this._kbnFilter = null;
     }
   }
 
+  /*
+   * Remove any user changes to value by resetting value to that as provided by Kibana filter pills
+   */
   reset() {
-    this._hasChanged = false;
     this._kbnFilter = null;
     this.value = this.filterManager.getValueFromFilterBar();
   }
 
+  /*
+   * Clear any filter on the field by setting the control value to undefined.
+   */
   clear() {
-    this.set(this.filterManager.getUnsetValue());
+    this.set();
```

Comment:
imo it'd be clearer to do `this.value = undefined;`  Can set ever be called outside of clear with `undefined`?  If not, we could throw an error in set if newValue is not defined (or when we eventually typescript we could tell then we don't support undefined), and then we don't have to check `this.hasValue()`, just always assume there is a value.

Reply:
We still need to support value being undefined to signal when the control is unset.

Reply:
I can't seem to trigger `set` being called with undefined outside of this clear function when playing around with the control. Can you tell me the instance you are referring to?  Or maybe we are misunderstanding each other.  I understand we need to support the value being undefined, I just mean writing the code like this would be clearer imo:

```
set(newValue) {
  this.value = newValue;
  this._kbnFilter = this.filterManager.createFilter(this.value);
}

clear() {
  this.value = undefined;
  this._kbnFilter = null;
}
```

Then when this supports typescript we can do something like `set(newValue: T)` instead of `set(newValue: T | undefined)`

Reply:
Here is a case where `hasValue` is needed. In the list control, when the list is cleared, set is called with `[]`. In this case, `_kbnFilter` needs to be set to null since there is not filter value. This logic could move into `filterManager.createFilter` if that makes things easier to understand.